products/oslogin/api.yaml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(nelsonjr): Make all Zone and Region resource ref
+
+--- !ruby/object:Api::Product
+name: OSLogin
+display_name: OS Login
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://oslogin.googleapis.com/v1/
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Identity and Access Management (IAM) API
+    url: https://console.cloud.google.com/apis/library/iam.googleapis.com/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+  - https://www.googleapis.com/auth/compute    
+objects:
+  - !ruby/object:Api::Resource
+    name: 'SSHPublicKey'
+    kind: user#sshPublicKeys
+    base_url: "users/{{user}}/sshPublicKeys/{{fingerprint}}"
```

Comment:
If you set `self_link` value to the same value as `base_url`, it will work correctly.

The default value for `self_link` is the `base_url` followed by `/{{name}}`.

Reply:
Sorry but i have the same error with the self_link. The generated is steel:
```
	if err := parseImportId([]string{
		"users/(?P<user>[^/]+)/sshPublicKeys/(?P<fingerprint>[^/]+)/(?P<name>[^/]+)",
		"(?P<user>[^/]+)/(?P<fingerprint>[^/]+)/(?P<name>[^/]+)",
```

And if i set :
```
    id_format: "users/{{user}}/sshPublicKeys/{{fingerprint}}"
    import_format: ["users/{{user}}/sshPublicKeys/{{fingerprint}}"]
```
The fingerprint is not available on the import test.

I am not sure that import is mandatory for this resources because the import keys are `user` and `fingerprint` but if you have the fingerprint, it's because you have the key.

do you agree if i deactivate imports?

Reply:
How can i use an import field that is not in resource properties?
Because to retrive the key we have to set the fingerprint and the user but to create we have to set the user and key content.

Reply:
I think I'd rather we keep the import. Terraform has the concept of managed/unmanaged resources, where import takes an unmanaged resource created out of band and begins managing it. 
It's possible users will have a key configured that they already have that they want Terraform to begin managing.

Fingerprint not being available in config / at creation time is fine- we can set it in the id in post_create. We've dealt with server-generated values a handful of times before. (Even though it's technically derivable, we can let the server generate it for us)

Reply:
Also whoops, mixed up which of the `self_link` and `identity` fields modify the id. `identity` doesn't make sense for this resource (normally the value would get appended to the base_url but that doesn't apply here), so we want to specify `id_format` and `import_format` explicitly in `terraform.yaml`.

Can you push the changes including the value you used above? That should be correct, and it's easier for me to take a look at what's going wrong once the generator runs.

Reply:
`import_format` and `id_format` added
And now i have this error on test:
```
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccOSLogin* -timeout 240m -ldflags="-X=github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccOSLoginSSHPublicKey_osLoginSshKeyProvidedUserExample
=== PAUSE TestAccOSLoginSSHPublicKey_osLoginSshKeyProvidedUserExample
=== CONT  TestAccOSLoginSSHPublicKey_osLoginSshKeyProvidedUserExample
    TestAccOSLoginSSHPublicKey_osLoginSshKeyProvidedUserExample: testing.go:635: Step 1 error: Error reading OSLoginSSHPublicKey "users/users/sshPublicKeys/<MY_EMAIL>m": googleapi: Error 403: End user credentials must match the user specified in the request.
--- FAIL: TestAccOSLoginSSHPublicKey_osLoginSshKeyProvidedUserExample (19.47s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-google/google	19.487s
FAIL
make: *** [GNUmakefile:19: testacc] Error 1

```

Reply:
For me the generator is good but the func `parseImportId` not work because the `fingerprint` is an output parameter.

Reply:
At this line: https://github.com/terraform-providers/terraform-provider-google/blob/5bdb7563939c4fd8479751ba23273a0f15583d68/google/import.go#L27

the id not contain the fingerprint

Reply:
Got it- that's actually happening because the `id` we're setting in Create doesn't include the fingerprint. The test runner uses the `id` as the import `id`. Added a suggestion to fix it.